### PR TITLE
Removed test launch dialogue , test cannot be accessed after EndDate

### PR DESCRIPTION
**Pull request by asif-khan17**
**Fixed Issues**

#Tes launch dialogue removed

**Files Changed**

TestConductRepository.cs - Added condition for test link cannot be accessed after EndDate
test-settings.component.ts- removed dialogue box

**Checks**

- [X] Naming Conventions
- [X] Server side code comments (proper English, grammar and no spelling mistakes)
- [X] Client side code comments (proper English, grammar and no spelling mistakes)
- [X] Unused variables, methods, blank spaces and name spaces. 
- [X] Optimal Code and code formatting
- [X] Commit History is proper, no merge is done. 
- [X] Commit/pull request messages should be proper to justify your feature
- [ ] All test cases are executed provided by tester.

**Known Issues**

- issue 1
- issue 2

**Comments**

NA

---
This change is [Reviewable](https://reviewable.io/reviews/promact/trappist/380)

### CODE REVIEW

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [systemjs.config.js#L5](https://github.com/Promact/trappist/blob/0bf6e319258dd60154093af895e3bd1c6ce9e334/Trappist/src/Promact.Trappist.Web/wwwroot/systemjs.config.js#L5): Remove the unused function parameter "global". [rule](http://server12:9005/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [systemjs.config.js#L5](https://github.com/Promact/trappist/blob/81ae9630385e64e20e35714b23c69ccacefedb88/Trappist/src/Promact.Trappist.Web/wwwroot/systemjs.config.js#L5): Remove the unused function parameter "global". [rule](http://server12:9005/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs [hunk 1]
```diff
@@ -85,10 +85,19 @@ public TestConductRepository(TrappistDbContext dbContext, ITestsRepository testR
...
         public async Task<bool> IsTestLinkExistForTestConductionAsync(string magicString)
-        {
-            return await (_dbContext.Test.AnyAsync(x => x.Link == magicString && !x.IsPaused));
+        {        
+            var testObject = await _dbContext.Test.FirstOrDefaultAsync(x => x.Link == magicString);
+            var currentDate = DateTime.UtcNow;
+            if (testObject != null)
+            {
+                var compareEndDate = DateTime.Compare(currentDate, testObject.EndDate);
+                if (testObject.IsPaused || (compareEndDate > 0))
```
Comment (rushi216) [inline Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs]: Refactor if else and return statements

File: Trappist/src/Promact.Trappist.Web/wwwroot/systemjs.config.js [hunk 2]
```diff
@@ -49,7 +49,7 @@
             'angular2-infinite-scroll': 'npm:angular2-infinite-scroll',
             'angular2-tinymce': 'npm:angular2-tinymce/dist',
             'tinymce': 'npm:tinymce',
-            'ngx-popover': 'npm:ngx-popover',
+            'ngx-popover': 'node_modules/ngx-popover',
```
Comment (rushi216) [inline Trappist/src/Promact.Trappist.Web/wwwroot/systemjs.config.js]: use npm:

File: Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs (hunk 1 above)
Comment (rushi216) [inline Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs]: It is not proper yet

File: Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs [hunk 3]
```diff
@@ -82,8 +82,13 @@ public TestConductRepository(TrappistDbContext dbContext, ITestsRepository testR
         }
 
         public async Task<bool> IsTestLinkExistForTestConductionAsync(string magicString)
-        {
-            return await (_dbContext.Test.AnyAsync(x => x.Link == magicString && !x.IsPaused));
+        {        
+            var testObject = await _dbContext.Test.FirstOrDefaultAsync(x => x.Link == magicString);
+            var currentDate = DateTime.UtcNow;
+            var compareEndDate = DateTime.Compare(currentDate, testObject.EndDate);
```
Comment (rushi216) [inline Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs]: it will throw error when testObject is null

**Pull request merged**